### PR TITLE
Reset basis function count

### PR DESCRIPTION
**Pull request by Laar**
The basis function count was not reset correctly when switching between different basis functions. 

When changing from CG to DG basis functions on the same mesh (probably does not happen in practice), the number of basis functions on Nodes, Edges and Faces was incorrect.

### CODE REVIEW

File: tests/unit/Base/BasisFunctionChange_UnitTest.cpp [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+// order of the quadrature).
+
+void initMPI() {
+#ifdef HPGEM_USE_MPI
+    int initialized;
+    MPI_Initialized(&initialized);
+    if (!initialized) {
+        MPI_Init(nullptr, nullptr);
+    }
+    std::atexit([]() { MPI_Finalize(); });
```
Comment (JensWehner) [inline tests/unit/Base/BasisFunctionChange_UnitTest.cpp:64]: nice

**Review by JensWehner — APPROVED** [review]

**Comment by JensWehner** [comment]
Good catch.

File: tests/unit/Base/BasisFunctionChange_UnitTest.cpp (hunk 1 above)
Comment (Laar) [reply tests/unit/Base/BasisFunctionChange_UnitTest.cpp:64]: ~~Inspiration~~ Copied from `CommandLineOptions`

**Comment by Laar** [comment]
I'll reformat the code this evening at home, easier than on the laptop.

**Pull request merged**